pkg/crypto/helpers_test.go

```diff
@@ -0,0 +1,35 @@
+package crypto
+
+import (
+	"testing"
+)
+
+func TestTopicFromValidCert(t *testing.T) {
+	certificate, _ := ReadPEMCertificateFile("testdata/mock_push_cert.pem")
+	pushTopic, err := TopicFromCert(certificate)
+
```

Comment:
remove the extra line. `if err != nil` should be grouped with the TopicFromCert call.

Reply:
Like this?
```
	if pushTopic, err := TopicFromCert(certificate); err != nil {
		t.Errorf("fail %s", err)
	} else {
		if have, want := pushTopic, "com.apple.mgmt.External.18a16429-886b-41f1-9c30-2bd04ae4fc37"; have != want {
			t.Errorf("have %d, want %d", have, want)
		}
	}
```

Not sure I like to nest the next test inside an else block, but the pushTopic variable will be scoped to the if block if do it like this.

Sorry for my n00b Go knowledge. I watched a Go crash course on high speed before starting coding. :)

Reply:
Sorry what I meant by this comment is to remove the newline between TopicFromCert and the `if err != nil` (line 10)

The actual code is fine.